src/element/typeChecks.ts

```diff
@@ -101,3 +103,30 @@ export const isExcalidrawElement = (element: any): boolean => {
     element?.type === "line"
   );
 };
+
+export const isRectangleElement = (
+  element: ExcalidrawElement | null,
+): element is ExcalidrawRectangleElement => {
+  return element != null && element.type === "rectangle";
+};
+
+export const hasBoundTextElement = (
+  element: ExcalidrawElement | null,
+): element is ExcalidrawBindableElement => {
+  return (
+    isBindableElement(element) &&
+    !!element.boundElements?.filter(({ type }) => type === "text").length
+  );
+};
+
+export const isBoundByContainer = (
+  element: ExcalidrawElement | null,
+): element is ExcalidrawTextElementWithContainer => {
+  return (
+    element !== null && isTextElement(element) && element.containerId !== null
+  );
+};
+
+export const getBoundTextElementId = (element: ExcalidrawBindableElement) => {
+  return element.boundElements!.filter((ele) => ele.type === "text")[0].id;
+};
```

Comment:
I can see that you're always using it in conjunction with `hasBoundTextElement()` to ensure type-safety. Meaning, this function on its own isn't type safe, and is tightly-coupled to `getBoundTextElementId()`.

I instead suggest you create a `getBoundTextElement(container: ExcalidrawElement | null): ExcalidrawTextElement | null` and in all the places you call the above 2 functions together, you call `getBoundTextElement()` instead, assigning the result to a variable and checking its existence. This will make it faster (a single checks/loops) and type-safe. There's also no guarantee that the bound text element actually exists (the `boundElements` array may contain invalid state in case of bugs): https://github.com/excalidraw/excalidraw/blob/160af4b641ebfe3ee36468246e4f86ad795297a7/src/element/resizeElements.ts#L166-L170

(note that this function shouldn't live in `typeChecks.ts` since it's not a boolean check.)